akka-actor-typed/src/main/scala/akka/actor/typed/delivery/ConsumerController.scala

```diff
@@ -124,6 +125,21 @@ object ConsumerController {
 
   final case class DeliverThenStop[A]() extends Command[A]
 
+  object SequencedMessage {
+
+    /**
+     * INTERNAL API
+     */
+    @InternalApi private[akka] def fromChunked[A](
+        producerId: String,
+        seqNr: SeqNr,
+        chunk: ChunkedMessage,
+        first: Boolean,
+        ack: Boolean,
+        producerController: ActorRef[ProducerControllerImpl.InternalCommand]): SequencedMessage[A] =
+      new SequencedMessage(producerId, seqNr, chunk.asInstanceOf[A], first, ack)(producerController)
```

Comment:
`A` is the user message type.
`A` is just a generic type and anything can be casted to that. The real type is checked when reading it. https://github.com/akka/akka/pull/28915/files#diff-e960aca9b7045b229b0aa4e2a07472b4R492

The alternative would be to have `SequencedMessage(message: Any)` to indicate that it can be different types, or use `message: Either[A, SequencedMessage[A]]`.

The latter would have a performance penalty.

I can try to change to Any if you prefer that. Using a `type MessageOrChunked = Any`

Reply:
I think Any would be more clear 👍

Reply:
this was a rather intrusive change, but it's better to not pretend. fixed in f4ca664f91